openmdao/jacobians/dictionary_jacobian.py

```diff
@@ -80,64 +80,79 @@ def _apply(self, d_inputs, d_outputs, d_residuals, mode):
         mode : str
             'fwd' or 'rev'.
         """
+        # avoid circular import
+        from openmdao.core.explicitcomponent import ExplicitComponent
+
         fwd = mode == 'fwd'
-        with self._system._unscaled_context(
-                outputs=[d_outputs], residuals=[d_residuals]):
+        system = self._system
+        d_res_names = d_residuals._names
+        d_out_names = d_outputs._names
+        d_inp_names = d_inputs._names
+        res_flat_views = d_residuals._views_flat
+        out_flat_views = d_outputs._views_flat
+        inp_flat_views = d_inputs._views_flat
+        np_add_at = np.add.at
+
+        with system._unscaled_context(outputs=[d_outputs], residuals=[d_residuals]):
```

Comment:
You added an unscaling here. Is this needed? How did this code work before if we were supposed to unscale here but did not?

Reply:
I didn't add an unscaling.  If you look above, it used to be two lines and I made it a single line.

Reply:
That is a relief.